### PR TITLE
Support download and ingest

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

File: src/context/test/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # attached with Common Clause Condition 1.0, found in the LICENSES directory.
 
 SET(CONTEXT_TEST_LIBS
+    $<TARGET_OBJECTS:common_ws_common_obj>
```
Comment (yixinglu) [inline src/context/test/CMakeLists.txt:7]: Please rebase your codebase, this compile error has been fixed.

File: src/planner/Admin.h [hunk 2]
```diff
@@ -417,10 +417,52 @@ class ShowListener final : public SingleInputNode {
 
 class Download final : public SingleInputNode {
```
Comment (Shylock-Hg) [inline src/planner/Admin.h:418]: SingleDependecyNode

File: src/planner/Admin.h [hunk 3]
```diff
@@ -417,10 +417,52 @@ class ShowListener final : public SingleInputNode {
...
+            hdfsPort_(hdfsPort),
+            hdfsPath_(hdfsPath) {}
+
+private:
+    std::string hdfsHost_;
+    int32_t hdfsPort_;
+    std::string hdfsPath_;
 };
 
 class Ingest final : public SingleInputNode {
```
Comment (Shylock-Hg) [inline src/planner/Admin.h:457]: Ditto.

File: src/executor/admin/DownloadExecutor.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/executor/admin/DownloadExecutor.cpp:1]: 2021

File: src/executor/admin/DownloadExecutor.h [hunk 5]
```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/executor/admin/DownloadExecutor.h:1]: ditto

File: src/validator/DownloadValidator.h [hunk 6]
```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/validator/DownloadValidator.h:1]: ditto

File: src/validator/DownloadValidator.cpp [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+#include "common/base/Base.h"
+
+#include "planner/Admin.h"
+#include "validator/DownloadValidator.h"
+#include "parser/MutateSentences.h"
+
+namespace nebula {
+namespace graph {
+
+Status DownloadValidator::toPlan() {
```
Comment (laura-ding) [inline src/validator/DownloadValidator.cpp:16]: you can put them into AdminValidator

File: src/planner/PlanNode.cpp [hunk 8]
```diff
@@ -247,6 +247,10 @@ const char* PlanNode::toString(PlanNode::Kind kind) {
             return "SignInTSService";
         case Kind::kSignOutTSService:
             return "SignOutTSService";
+        case Kind::kDownload:
+            return "kDownload";
```
Comment (laura-ding) [inline src/planner/PlanNode.cpp:251]: moidfy `"kDownload"` to `"Download"`

File: src/planner/PlanNode.cpp [hunk 9]
```diff
@@ -247,6 +247,10 @@ const char* PlanNode::toString(PlanNode::Kind kind) {
...
+        case Kind::kIngest:
+            return "kIngest";
```
Comment (laura-ding) [inline src/planner/PlanNode.cpp:253]: ditto

File: src/executor/admin/DownloadExecutor.cpp (hunk 4 above)
Comment (darionyaphet) [reply src/executor/admin/DownloadExecutor.cpp]: fixed ~

**Review by yixinglu — APPROVED** [review]

**Review by laura-ding — APPROVED** [review]

**Pull request merged**